File: 1038/1038.java

class Solution {
    int sum=0;
    public TreeNode bstToGst(TreeNode root){if(root.right!=null)bstToGst(root.right);sum+=root.val;root.val=sum;if(root.left!=null)bstToGst(root.left);return root;}
}

File: 1038/1038.py
class Solution:
    def binary_search_tree_to_greater_sum_tree(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.binary_search_tree_to_greater_sum_tree(root.left), self.binary_search_tree_to_greater_sum_tree(root.right))